File: dsa/graph/226_Invert_Binary_Tree.java
package graph;


//  * Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public TreeNode invertTree(TreeNode root) {
        this.invertTreeHelper(root);
        return root;
    }

    public void invertTreeHelper(TreeNode root){
        if (root == null){
            return;
        }
        TreeNode tempNode = root.left;
        root.left = root.right;
        root.right = tempNode;
        this.invertTreeHelper(root.left);
        this.invertTreeHelper(root.right);
    }
}

File: dsa/10+11.graph/226_Invert_Binary_Tree.py
import typing as t
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def invertTree(self, root: t.Optional[TreeNode]) -> t.Optional[TreeNode]:
        self.invertTreeHelper(root)
        return root
    
    def invertTreeHelper(self, root: t.Optional[TreeNode]) -> None:
        if not root:
            return
        
        tempNode: TreeNode = root.left
        root.left = root.right
        root.right = tempNode
        
        self.invertTreeHelper(root.left)
        self.invertTreeHelper(root.right)
        